lib/responses.js

```diff
@@ -503,13 +503,15 @@ function listWithPublishedVersions(req, res) {
  * @param {object} res
  */
 function listAsReferencedObjects(req, res) {
-  expectJSON(function () {
-    return db.pipeToPromise(db.list({
+  expectJSON(() => {
+    var promise = db.pipeToPromise(db.list({
       keys: true,
       values: true,
       isArray: true,
       prefix: req.uri
-    })).then(JSON.parse).then(convertDBListToReferenceObjects);
+    }));
+
+    return promise.then(JSON.parse).then(convertDBListToReferenceObjects);
```

Comment:
does `pipeToPromise` still return strings of JSON?

Reply:
Unfortunately yes because the stream requires strings to operate, so we have to parse it out. I've done a little refactor on this and will push it up soon. The only thing using method is scheduling, so when that gets pulled out this is just going to move over